### PR TITLE
ccw4102

### PR DESCRIPTION
**Pull request by nke001**
fix gh-4102

### CODE REVIEW

**Comment by nke001** [comment]
@nouiz

**Comment by nouiz** [comment]
/home/travis/build/Theano/Theano/theano/compile/profilemode.py:705:26: E211 whitespace before '('
/home/travis/build/Theano/Theano/theano/compile/profilemode.py:708:26: E211 whitespace before '('

**Comment by nouiz** [comment]
travis still fail.

File: theano/compile/profilemode.py [hunk 1]
```diff
@@ -693,6 +693,22 @@ def exp_float32_op(op):
...
 
+        # tip 6
+        import theano.sandbox.cuda as cuda
+        from theano.tensor.nnet import LogSoftmax
+        import theano.tensor.signal.pool as pool
+
+        for a, t in iteritems(apply_time):
+            node = a[1]
+            if (isinstance(node.op, pool.Pool)):
+                if not cuda.dnn.dnn_available():
```
Comment (nouiz) [inline theano/compile/profilemode.py]: Also do the check for the new back-end
import theano.gpuarray.dnn
if not not cuda.dnn.dnn_available() or not theano.gpuarray.dnn.dnn_present()

idem bellow.

Comment (nouiz) [inline theano/compile/profilemode.py]: You didn't do that change.

File: theano/compile/profilemode.py [hunk 2]
```diff
@@ -693,6 +693,22 @@ def exp_float32_op(op):
                           " generator supported on the GPU.")
                 break
 
+        # tip 6
+        import theano.sandbox.cuda as cuda
+        from theano.tensor.nnet import LogSoftmax
+        import theano.tensor.signal.pool as pool
+
```
Comment (nke001) [inline theano/compile/profilemode.py]: It's done :)

**Comment by nke001** [comment]
@nouiz Ive made all the changes, and it has passed the tests. i think it can be merged now.

**Comment by nouiz** [comment]
thanks

**Pull request merged**